src/System.Reflection.Metadata/tests/Metadata/MetadataReaderTests.cs

```diff
@@ -2503,5 +2503,143 @@ private static FileStream GetTemporaryAssemblyLargeEnoughToBeMemoryMapped()
+                {
+                    PEMemoryBlock metadata = peReader.GetMetadata();
+                    peReader.GetMetadata();
+                    offsetToMetadata = peReader.PEHeaders.MetadataStartOffset;
+                    offsetToMetadataSize = peReader.PEHeaders.CorHeaderStartOffset + 12;
+                    offsetToModuleTable = offsetToMetadata + peReader.GetMetadataReader().GetTableMetadataOffset(TableIndex.Module);
+
+                    // skip root header
+                    BlobReader blobReader = new BlobReader(metadata.Pointer, metadata.Length);
+                    blobReader.ReadUInt32(); // signature
+                    blobReader.ReadUInt16(); // major version
+                    blobReader.ReadUInt16(); // minor version
+                    blobReader.ReadUInt32(); // reserved
+                    int versionStringSize = blobReader.ReadInt32();
+                    blobReader.SkipBytes(versionStringSize);
+
+                    // read stream headers to collect offsets and sizes to adjust later
+                    blobReader.ReadUInt16(); // reserved
+                    int streamCount = blobReader.ReadInt16();
+                    streamHeaders = new StreamHeaderInfo[streamCount];
+
+                    for (int i = 0; i < streamCount; i++)
+                    {
+                        streamHeaders[i].OffsetToOffset = offsetToMetadata + blobReader.Offset;
+                        streamHeaders[i].Offset = blobReader.ReadInt32();
+                        streamHeaders[i].OffsetToSize = offsetToMetadata + blobReader.Offset;
+                        streamHeaders[i].Size = blobReader.ReadInt32();
+
+                        string name = blobReader.ReadUtf8NullTerminated();
+                        if (name == "#~")
+                        {
+                            tableStreamIndex = i;
+                        }
+
+                        blobReader.Align(4);
+                    }
+                }
+            }
+
+            const int sizeOfExtraData = 4;
+            int offsetToTableStream = offsetToMetadata + streamHeaders[tableStreamIndex].Offset;
+            int offsetToHeapSizeFlags = offsetToTableStream + 6;
+
+            // copy unobfuscated to obfuscated, leaving room for 4 bytes of data before right before the module table.
+            byte[] obfuscated = new byte[unobfuscated.Length + sizeOfExtraData];
+            Array.Copy(unobfuscated, 0, obfuscated, 0, offsetToModuleTable);
+            Array.Copy(unobfuscated, offsetToModuleTable, obfuscated, offsetToModuleTable + sizeOfExtraData, unobfuscated.Length - offsetToModuleTable);
+
+            fixed (byte* ptr = obfuscated)
+            {
+                // increase size of metadata
+                *(int*)(ptr + offsetToMetadataSize) += sizeOfExtraData;
+
+                // increase size of table stream
+                *((int*)(ptr + streamHeaders[tableStreamIndex].OffsetToSize)) += sizeOfExtraData;
```

Comment:
Self-nit: be consistent on parens here and below. I'll remove the outer set from 2 of 3 cases like this.